### ISSUE TITLE
A suggestion to reduce the number of vector allocations in getRotation of Matrix3.js

### ISSUE THREAD

**Issue by bereaphysics** [phetsims/dot#23]
While looking up the source of allocations in charges and fields, a sizable number of them are involved in the rotation of the node (using setRotation (which itself invoke getRotation)).

In any case, if you carry out the math explicitly, you find that getRotation in Matrix3.js

````
    // angle in radians for the 2d rotation from this matrix, between pi, -pi
    getRotation: function() {
      var transformedVector = this.timesVector2( dot.Vector2.X_UNIT ).minus( this.timesVector2( dot.Vector2.ZERO ) );
      return Math.atan2( transformedVector.y, transformedVector.x );
    },

````

can be replaced by

````
    // angle in radians for the 2d rotation from this matrix, between pi, -pi
    getRotation: function() {
      return Math.atan2( this.m10(), this.m00() );
````

**Comment by jonathanolson** [comment]
Ah yes, ```M * v( 1, 0, 1 ) - M * v( 0, 0, 1 )``` (homogeneous vectors) distributes to ```M * v( 1, 0, 0 )```. Thanks for catching that!

I'll make the fix shortly, and it should free up a lot of those vector allocations you were seeing.

**Comment by jonathanolson** [comment]
Fixed in 32a2f0072ffaac161e5071290f1658235638b9bf, forgot to tag it.

Thanks!
